commands/init.go

```diff
@@ -27,70 +41,174 @@ var initCmd = &cmds.Command{
+		peerKeyFile, _ := req.Options[PeerKeyFile].(string)
+		initopts, err := getNodeInitOpts(autoSealIntervalSeconds, peerKeyFile)
+		if err != nil {
+			return err
 		}
 
-		return GetAPI(env).Daemon().Init(
-			req.Context,
-			api.RepoDir(repoDir),
-			api.GenesisFile(genesisFile),
-			api.PeerKeyFile(peerKeyFile),
-			api.WithMiner(withMiner),
-			api.DevnetTest(devnetTest),
-			api.DevnetNightly(devnetNightly),
-			api.DevnetUser(devnetUser),
-			api.AutoSealIntervalSeconds(autoSealIntervalSeconds),
-			api.DefaultAddress(defaultAddress),
-		)
+		// TODO: don't create the repo if this fails
+		return node.Init(req.Context, rep, genesisFile, initopts...)
 	},
 	Encoders: cmds.EncoderMap{
 		cmds.Text: cmds.MakeEncoder(initTextEncoder),
 	},
 }
 
+func getConfigFromOptions(options cmdkit.OptMap) (*config.Config, error) {
+	newConfig := config.NewDefaultConfig()
+
+	if m, ok := options[WithMiner].(string); ok {
+		var err error
+		newConfig.Mining.MinerAddress, err = address.NewFromString(m)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if m, ok := options[DefaultAddress].(string); ok {
+		var err error
+		newConfig.Wallet.DefaultAddress, err = address.NewFromString(m)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	devnetTest, _ := options[DevnetTest].(bool)
+	devnetNightly, _ := options[DevnetNightly].(bool)
+	devnetUser, _ := options[DevnetUser].(bool)
+	if (devnetTest && devnetNightly) || (devnetTest && devnetUser) || (devnetNightly && devnetUser) {
+		return nil, fmt.Errorf(`cannot specify more than one "devnet-" option`)
+	}
+
+	// Setup devnet specific config options.
+	if devnetTest || devnetNightly || devnetUser {
+		newConfig.Bootstrap.MinPeerThreshold = 1
+		newConfig.Bootstrap.Period = "10s"
+	}
+
+	// Setup devnet test specific config options.
+	if devnetTest {
+		newConfig.Bootstrap.Addresses = fixtures.DevnetTestBootstrapAddrs
+		newConfig.Net = "devnet-test"
+	}
+
+	// Setup devnet nightly specific config options.
+	if devnetNightly {
+		newConfig.Bootstrap.Addresses = fixtures.DevnetNightlyBootstrapAddrs
+		newConfig.Net = "devnet-nightly"
+	}
+
+	// Setup devnet user specific config options.
+	if devnetUser {
+		newConfig.Bootstrap.Addresses = fixtures.DevnetUserBootstrapAddrs
+		newConfig.Net = "devnet-user"
+	}
+
+	return newConfig, nil
+}
+
 func initTextEncoder(req *cmds.Request, w io.Writer, val interface{}) error {
 	_, err := fmt.Fprintf(w, val.(string))
 	return err
 }
 
-func getRepoDir(req *cmds.Request) string {
-	envdir := os.Getenv("FIL_PATH")
+func loadGenesis(ctx context.Context, rep repo.Repo, sourceName string) (consensus.GenesisInitFunc, error) {
+	if sourceName == "" {
+		return consensus.DefaultGenesis, nil
+	}
+
+	sourceURL, err := url.Parse(sourceName)
+	if err != nil {
+		return nil, fmt.Errorf("invalid filepath or URL for genesis file: %s", sourceURL)
+	}
+
+	var source io.ReadCloser
+	if sourceURL.Scheme == "http" || sourceURL.Scheme == "https" {
```

Comment:
when this merges could you please file an issue to remove http support, labeling it i-security. does not have to be prioritized, can just sit in the backlog